packages/Telephony/Actions/SerialNumberInput.cs

```diff
@@ -142,56 +164,80 @@ public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext d
                         return await PromptUserAsync(dc, cancellationToken).ConfigureAwait(false);
                 }
 
-                if (snp.PatternLength == result.Length)
+                if (batchLength == result.Length)
                 {
                     //If so, save it to the output property and end the dialog
                     //Get property from expression
-                    string property = this.Property?.GetValue(dc.State);
+                    var property = Property?.GetValue(dc.State);
                     dc.State.SetValue(property, result);
                     return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
                 }
                 else
                 {
                     dc.State.SetValue(AggregationDialogMemory, result);
-                    await this.SendActivityMessageAsync(this.ContinuePrompt, dc, cancellationToken).ConfigureAwait(false);
+                    await SendActivityMessageAsync(ContinuePrompt, dc, cancellationToken).ConfigureAwait(false);
                     return new DialogTurnResult(DialogTurnStatus.Waiting);
                 }
             }
-            
-            //append the message to the aggregation memory state
+
+            //If input is longer than pattern length and wasn't handled as an interruption, silently disregard input.
+            var userInput = dc.Context.Activity.Text;
+
+            if (userInput.Length > batchLength)
+            {
+                return new DialogTurnResult(DialogTurnStatus.Waiting);
```

Comment:
>ialogTurnResult( [](http://example.com/codeflow?start=28&length=16)

How do they get out of this state? Wouldn't it just wait for more input and always be longer?

Reply:
This check is along the lines of `if the user says 233 when the bot is expecting a batch length of 2 (e.g., 23 or 23), wait for the user to say the correct length.`

I'm not entirely happy with the approach I went with here, as it's a very by-the-rails experience happy path.

Ideally, there'd be some logic where C1 can have their bot say "This is what you've currently provided `xyz`" after _n_-amount of attempts to provide what **C2** thinks is correct, but what the bot is not equipped to handle in the current node.

These thoughts are briefly covered in a TODO on lines [201-207](https://github.com/microsoft/botframework-components/pull/1389/files/a043665981ef4117e529a952edb52b59beab7915#diff-5e26f7355c395f29b1b2eaa57b5ec940cbe84ec12647c8e311cd30dde9c20f23R201-R207). I didn't complete exploring that functionality, but I strongly believe that in any voice-first bot, these are scenarios and features that need to be explored and haven't been done so by the code in the SDK or the Telephony extensions.

Reply:
I don't think the user will _ever_ figure out that they said too much and repeat exactly what's needed. At that point they will be most likely shouting "agent, agent!" or just hang up. I would suggest getting C2 out of this state as quickly as possible (either take what you need from the utterance to make the full length and ignore the rest, or just error out) and recommend the bot to always confirm the selection with the option to re-enter as a best practice.